resources/service_certmapping.rb

```diff
@@ -17,24 +17,98 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-default_action :configure
-actions :configure, :delete
-
-attribute :enabled,  default: true,  kind_of: [TrueClass, FalseClass]
-attribute :issuer,   required: true, kind_of: String
-attribute :password, required: true, kind_of: String
-attribute :subject,  default: '*',   kind_of: String
-attribute :uri,      default: '*',   kind_of: String
-attribute :username, required: true, kind_of: String
-
-def path
-  @path ||= "winrm/config/service/certmapping?Issuer=#{issuer}+Subject=#{subject}+URI=#{uri}"
+
+property :enabled,  [TrueClass, FalseClass], default: true
+property :issuer,   String, required: true, identity: true
+property :password, String, required: true
+property :subject,  String, default: '*', identity: true
+property :uri,      String, default: '*', identity: true
+property :username, String, required: true
+
+include ::WinrmConfig::ServiceCertmappingHelpers
+
+load_current_value do |_desired_state|
+  begin
+    winrm_data = winrm_get
+  rescue StandardError => e
+    ::Chef::Log.warn("An error occured while retrieving certmapping: #{e.message}")
+  end
+  current_value_does_not_exist! if winrm_data.nil? || winrm_data.empty?
+
+  username winrm_data['UserName']
+  enabled(winrm_data['Enabled'] != 'false')
+end
```

Comment:
We do not need to set `issuer`, `subject`, `uri` ? Like with `hydrate` previously ?

Reply:
No we don't since these property are marked as `identity`